app/models/order.rb

```diff
@@ -4,6 +4,7 @@ class Order <ApplicationRecord
   belongs_to :user
   has_many :item_orders
   has_many :items, through: :item_orders
+  has_many :merchants, through: :items
```

Comment:
Same as above.  An order should already be able to get to merchants through items